File: 20.有效的括号.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s=="")  
            return true;
        stack<char> st;
        st.push(s[0]);
        for(int i=1; i<s.length(); ++i){
            if(s[i] == '(' || s[i] == '{' || s[i] == '[')
                st.push(s[i]);
            if(s[i] == ')'){
                if(st.empty() || st.top()!='(')
                    return false;
                st.pop();
            }
            if(s[i] == '}'){
                if(st.empty() || st.top()!='{')
                    return false;
                st.pop();
            }
            if(s[i] == ']'){
                if(st.empty() || st.top()!='[')
                    return false;
                st.pop();
            }
        }
        if(st.empty())
            return true;
        return false;
    }
};

File: 20.有效的括号.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    let stack = [];
    for (let c of s) {
        if (c === '(' || c === '[' || c === '{') {
            stack.push(c);
            continue;
        }
        if (stack.length === 0 || c === ')' && stack[stack.length - 1] !== '(' || c === ']' && stack[stack.length - 1] !== '[' || c === '}' && stack[stack.length - 1] !== '{') {
            return false;
        } else {
            stack.pop();
        }
    }
    return stack.length === 0;
};